shuup/admin/modules/media/views.py

```diff
@@ -130,33 +186,39 @@ def post(self, request, *args, **kwargs):
             return JsonResponse({"error": "unknown action %s" % action})
 
     def handle_get_folders(self, data):
-        root_folders = get_cached_trees(Folder._tree_manager.all())
+        shop = get_shop(self.request)
+        root_folders = get_cached_trees(Folder._tree_manager.filter(_get_folder_query_filter(shop)))
         return JsonResponse({"rootFolder": _filer_folder_to_json_dict(None, root_folders)})
 
     def handle_post_new_folder(self, data):
+        shop = get_shop(self.request)
         parent_id = int(data.get("parent", 0))
         if parent_id > 0:
-            parent = Folder.objects.get(pk=parent_id)
+            parent = _get_folder_query(shop).get(pk=parent_id)
         else:
             parent = None
+
         name = data["name"]
         folder = Folder.objects.create(name=name)
         if parent:
             folder.move_to(parent, "last-child")
             folder.save()
+
+        _ensure_media_folder(shop, folder)
         return JsonResponse({"success": True, "folder": _filer_folder_to_json_dict(folder, ())})
 
     def handle_get_folder(self, data):
+        shop = get_shop(self.request)
         try:
             folder_id = int(data.get("id", 0))
             if folder_id:
-                folder = Folder.objects.get(pk=folder_id)
-                subfolders = folder.get_children()
-                files = folder.files
+                folder = _get_folder_query(shop).get(pk=folder_id)
+                subfolders = folder.get_children().filter(_get_folder_query_filter(shop))
+                files = _get_file_query(shop, folder)
             else:
                 folder = None
-                subfolders = Folder.objects.filter(parent=None)
-                files = File.objects.filter(folder=None)
+                subfolders = _get_folder_query(shop).filter(parent=None)
+                files = _get_file_query(shop).filter(folder=None)
```

Comment:
check below the line `"error": "Folder does not exist"`
I guess you could make that a translated string right (do not use `ugettext_lazy`)? :sweat_smile: 
Maybe check around this files if there is any other error string to be translated. I guess they are returned to the users.

Reply:
If these untranslated "not found" errors starts to happen then let's add the translation separately. These are catched at post already.

Reply:
There is already try catch around these queries.